Mage.Sets/src/mage/cards/p/ProgenitorsIcon.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public ProgenitorsIconAsThoughEffect copy() {
+        return new ProgenitorsIconAsThoughEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID sourceId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        SubType subType = ChooseCreatureTypeEffect.getChosenCreatureType(source.getSourceId(), game);
+
+        if (!ProgenitorsIconWatcher.checkPlayerCast(affectedControllerId, subType, game)) {
+            discard();
+            return false;
+        }
+        Card card = game.getCard(sourceId);
+        if (card == null){
+            return false;
+        }
+
+        return card.getSubtype().contains(subType);
+    }
+}
+
+class ProgenitorsIconWatcher extends Watcher {
+
+    // Handles multiple instances of this same card choosing the same or different subtypes for the same player
+    private final Map<UUID, Set<SubType>> playerSubTypesChosen = new HashMap<>();
+
+    public ProgenitorsIconWatcher() {
+        super(WatcherScope.GAME);
+    }
+
+    @Override
+    public void watch(GameEvent event, Game game) {
+        if (event.getType() == GameEvent.EventType.SPELL_CAST){
+
+            Spell spell = game.getSpellOrLKIStack(event.getSourceId());
+            Set<SubType> chosenSubTypes = playerSubTypesChosen.get(event.getPlayerId());
+
+            if (spell == null || chosenSubTypes == null){
+                return;
+            }
+
+            // Remove all matching subtypes. If you tap two Progenitor's Icons for flash applying to Human and Goblin
+            // subtypes, then cast a Human Goblin with flash, both instances of Progenitor's Icon abilities get used.
+            for (SubType subType : spell.getSubtype()){
+                chosenSubTypes.remove(subType);
+            }
+        }
+    }
+
+    public static void addPlayer(UUID playerId, SubType subtype, Game game) {
+        Map<UUID, Set<SubType>> playerSubTypesChosen =
+                game.getState().getWatcher(ProgenitorsIconWatcher.class).playerSubTypesChosen;
+        // Add playerId to map linked to empty list if not already there
+        playerSubTypesChosen.computeIfAbsent(playerId, x -> new HashSet<>());
+        // Will not add a subtype twice
+        playerSubTypesChosen.get(playerId).add(subtype);
+    }
+
+    public static boolean checkPlayerCast(UUID playerId, SubType subtype, Game game) {
+        Map<UUID, Set<SubType>> playerSubTypesChosen =
+                game.getState().getWatcher(ProgenitorsIconWatcher.class).playerSubTypesChosen;
+        // If a player has tapped this for its effect,
+        // and at least one of those tapped Progenitor's Icons has this subtype selected
+        return playerSubTypesChosen.getOrDefault(playerId, new HashSet<>()).contains(subtype);
```

Comment:
In this case the default is never stored, only used to check contains, so better to use `Collections.emptySet()` rather than `new HashSet<>()`